src/lib/log-forwarding.js

```diff
@@ -0,0 +1,253 @@
+      const oldChecksum = fs.readFileSync(process.cwd() + '/tmp/' + CHECKSUM_FILE).toString()
+      const config = this.getLocalConfigWithSecrets()
+      const newChecksum = getChecksum(config)
+      return oldChecksum !== newChecksum
+    } else {
+      return true
+    }
+  }
+
+  async updateServerConfig (lfConfig) {
+    await this.logForwarding.setDestination(lfConfig.getDestination(), lfConfig.getSettings())
+    const checksum = getChecksum(lfConfig)
+    fs.ensureDirSync(process.cwd() + '/tmp')
+    fs.writeFile(process.cwd() + '/tmp/' + CHECKSUM_FILE, checksum, { flags: 'w' })
+  }
+}
+
+class LogForwardingConfig {
+  constructor (destination, settings) {
+    this.destination = destination
+    this.settings = settings
+  }
+
+  getDestination () {
+    return this.destination
+  }
+
+  getSettings () {
+    return this.settings
+  }
+
+  isDefined () {
+    return this.destination !== undefined
+  }
+
+  isDefault () {
+    return !this.isDefined() || this.getDestination() === 'adobe_io_runtime'
+  }
+
+  isEqual (config) {
+    return (this.isDefault() && config.isDefault()) ||
+      (this.destination === config.getDestination() && shallowEqual(this.settings, config.settings))
+  }
+}
+
+/**
+ * Init Log Forwarding
+ *
+ * @param {object} aioConfig aio Config
+ * @returns {Promise<LogForwarding>} Log Forwarding
+ */
+async function init (aioConfig) {
+  const lf = new LogForwarding(aioConfig)
+  return await lf.init()
+}
+
+/**
+ * Get Runtime Log Forwarding
+ *
+ * @param {object} rtConfig Runtime config
+ * @returns {Promise<LogForwarding>} Log Forwarding
+ */
+async function getRTLogForwarding (rtConfig) {
+  const rt = await rtLib.init({
+    ...rtConfig,
+    api_key: rtConfig.auth
+  })
+  return rt.logForwarding
+}
+
+/**
+ * Compare to log forwarding configs
+ *
+ * @param {LogForwardingConfig} config1 Config
+ * @param {LogForwardingConfig} config2 Config
+ * @returns {boolean} Are configs equal
+ */
+function shallowEqual (config1, config2) {
+  // updated_at exists on server only and does not impact actual configuration
+  const keys1 = Object.keys(config1).filter(e => e !== 'updated_at')
+  const keys2 = Object.keys(config2).filter(e => e !== 'updated_at')
+  if (keys1.length !== keys2.length) {
+    return false
+  }
+  for (const key of keys1) {
+    if (config1[key] !== config2[key]) {
+      return false
+    }
+  }
+  return true
+}
+
+/**
+ * Convert JSON config to Log Forwarding object
+ *
+ * @param {string} configJson Config in JSON format
+ * @returns {LogForwardingConfig} Config
+ */
+function convertToConfigObject (configJson) {
+  let destination
+  let settings
+
+  if (configJson !== undefined) {
+    const destinations = Object.keys(configJson)
+    if (destinations.length === 1) {
+      destination = destinations[0]
+      settings = configJson[destination]
+    } else {
+      throw new Error(`Configuration has ${destinations.length} destinations. Exactly one must be defined.`)
```

Comment:
I am not sure why this check is required, how would we get a config object with multiple destinations?  Seems like that would be an error in code.

Reply:
If somebody changes config manually. We can't guarantee this doesn't happen, so there should be intentional behavior in this case. But in normal scenario this shouldn't happen.